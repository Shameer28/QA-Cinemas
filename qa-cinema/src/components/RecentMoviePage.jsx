import axios from "axios";
import PosterLink from "./props/PosterLink";

import ListMovies from "./props/ListMovies"

const RecentMoviesPage = () => {
	const sortFunc = (a, b) => {
		return a.releaseDate > b.releaseDate;
	}

    return (<div>
        <h1>Recent Movies</h1>
        <ListMovies url="http://localhost:3000/movies/getAll" sort = {sortFunc}/>
    </div>);
}
 
export default RecentMoviesPage;