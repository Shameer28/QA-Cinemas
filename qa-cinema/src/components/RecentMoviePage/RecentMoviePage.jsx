import { Container } from "react-bootstrap";
import ListMovies from "./../sharedcomponents/ListMovies"

const RecentMoviesPage = () => {
    const sortFunc = (a, b) => {

		console.log(a.releaseDate, b.releaseDate)

        return a.releaseDate < b.releaseDate;
    }

    return (
        <div class="background">
            <br></br>
            <Container>
                <h2 style={{ color: "white", fontWeight: "400", fontfamily: "isonormregular, sans-serif", letterSpacing: ".15em", textTransform: "uppercase", lineHeight: "1.1" }}>New Releases</h2>
                <br></br>
                <ListMovies sort={sortFunc} />
            </Container>
        </div>);
}
export default RecentMoviesPage;
