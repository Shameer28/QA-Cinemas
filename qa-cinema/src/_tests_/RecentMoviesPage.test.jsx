import RecentMoviePage from '../components/RecentMoviePage/RecentMoviePage'
import { create } from 'react-test-renderer';


describe(`Recent Movie Page testing`, () => {

    it(`Should match the snapshot`, () => {
        const instance = create(<RecentMoviePage/>).toJSON();
        expect(instance).toMatchSnapshot();
    })
})