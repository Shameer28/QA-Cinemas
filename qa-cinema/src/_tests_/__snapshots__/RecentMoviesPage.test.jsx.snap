// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recent Movie Page testing Should match the snapshot 1`] = `
<div
  class="background"
>
  <br />
  <div
    className="container"
  >
    <h2
      style={
        Object {
          "color": "white",
          "fontWeight": "400",
          "fontfamily": "isonormregular, sans-serif",
          "letterSpacing": ".15em",
          "lineHeight": "1.1",
          "textTransform": "uppercase",
        }
      }
    >
      Recent Movies
    </h2>
    <br />
    <p>
      Loading
    </p>
  </div>
</div>
`;
